src/lib/viewers/image/ImageViewer.js

```diff
@@ -126,79 +143,57 @@ class ImageViewer extends ImageBaseViewer {
     zoom(type) {
         let newWidth;
         let newHeight;
-        const imageCurrentDimensions = this.imageEl.getBoundingClientRect(); // Getting bounding rect does not ignore transforms / rotates
-        const { height, width } = imageCurrentDimensions;
-        const aspect = width / height;
-        const viewport = {
-            width: this.wrapperEl.clientWidth - IMAGE_PADDING,
-            height: this.wrapperEl.clientHeight - IMAGE_PADDING
-        };
-
-        // For multi page tifs, we always modify the width, since its essentially a DIV and not IMG tag.
-        // For images that are wider than taller we use width. For images that are taller than wider, we use height.
-        const modifyWidthInsteadOfHeight = aspect >= 1;
+        let height;
+        let width;
 
         // From this point on, only 1 dimension will be modified. Either it will be width or it will be height.
         // The other one will remain null and eventually get cleared out. The image should automatically use the proper value
         // for the dimension that was cleared out.
-        switch (type) {
-            case 'in':
+        if (type === 'in' || type === 'out') {
+            const imageCurrentDimensions = this.imageEl.getBoundingClientRect(); // Getting bounding rect does not ignore transforms / rotates
+            ({ height, width } = imageCurrentDimensions);
+
+            const modifyWidthInsteadOfHeight = this.getModifyWidthInsteadOfHeight(width, height);
+            if (modifyWidthInsteadOfHeight) {
+                newWidth = type === 'in' ? width * IMAGE_ZOOM_SCALE : width / IMAGE_ZOOM_SCALE;
+            } else {
+                newHeight = type === 'in' ? height * IMAGE_ZOOM_SCALE : (newHeight = height / IMAGE_ZOOM_SCALE);
```

Comment:
I think you can remove the assignment as the last component of the ternary.

`newHeight = type === 'in' ? height * IMAGE_ZOOM_SCALE : height / IMAGE_ZOOM_SCALE;`

Reply:
Good catch, that was what I meant to put there to match the newWidth expression